### PR TITLE
Refactor/add more unit test

### PR DESCRIPTION
**Pull request by musamusa**

### CODE REVIEW

**Comment by jjal** [comment]
tests are good add more tests (e.g. for multi stockout save)

File: test/spec/controllers/stock-out-broadcast.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      _stockOutBroadcastFactory_, _notificationService_, _$q_){
+    scope = {};
+    state = $state;
+    $q = _$q_;
+    notificationService = _notificationService_;
+    appConfig = appConfigMock;
+    _i18n = i18n;
+    stockOutBroadcastFactory = _stockOutBroadcastFactory_;
+    stateParams = $stateParams;
+    stateParams.productList = '0930b906-4802-4a65-8516-057bd839db3e,939d5e05-2aa4-4883-9246-35c60dfa06a5';
```
Comment (jjal) [inline test/spec/controllers/stock-out-broadcast.js:19]: magic numbers

File: test/spec/controllers/stock-out-broadcast.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+      appConfig: appConfig,
+      i18n: _i18n,
+      stateParams: stateParams,
+      stockOutBroadcastFactory: stockOutBroadcastFactory,
+      facilityStockListProductTypes: productTypeMock,
+      notificationService: notificationService
+    });
+  }));
+
+  it('should covert convert uuid from url to array', function(){
```
Comment (jjal) [inline test/spec/controllers/stock-out-broadcast.js:33]: i don't understand this case

File: test/spec/controllers/stock-out-broadcast.js (hunk 1 above)
Comment (musamusa) [inline test/spec/controllers/stock-out-broadcast.js:19]: not really. its a comma separated uuids in I pass to the url, which is then converted to a list. this '0930b906-4802-4a65-8516-057bd839db3e,939d5e05-2aa4-4883-9246-35c60dfa06a5'
becomes 
[
"0930b906-4802-4a65-8516-057bd839db3e",
 "939d5e05-2aa4-4883-9246-35c60dfa06a5"
]

File: test/spec/controllers/stock-out-broadcast.js (hunk 2 above)
Comment (musamusa) [inline test/spec/controllers/stock-out-broadcast.js:33]: Oops!

Comment (musamusa) [inline test/spec/controllers/stock-out-broadcast.js:33]: it was to suppose to be ' Should convert url params to array' as stated above

**Pull request merged**